python/ray/workflow/workflow_state_from_dag.py

```diff
@@ -115,17 +105,18 @@ def _node_visitor(node: Any) -> Any:
                 else:
                     catch_exceptions = False
 
+            # We do not need to check the validness of bound options, because
+            # Ray options already check them for us.
             max_retries = bound_options.get("max_retries", 3)
-            if not isinstance(max_retries, int) or max_retries < -1:
-                raise ValueError(
-                    "'max_retries' only accepts 0, -1 or a positive integer."
-                )
+            # 'retry_exceptions' is True by default in Ray Workflow, but it is
```

Comment:
Any reason for this? SHouldn't it be aligned with normal ray?

Reply:
Also in ray, retry_exceptions can be a list of Exceptions.

Reply:
@iycheng The PR for list of exceptions is not ready: https://github.com/ray-project/ray/pull/26449.

Would you think that by default `max_retries` does not work for exceptions in workflow (due to `retry_exceptions=False` by default in Ray)? I am ok with both semantics.

Reply:
I feel a little bit confused. If max_retries=3 and retry_exceptions=False, would we retry anything?

I think in ray, it'll retry RaySystem errors.

Reply:
I'll just align with Ray semantics then anyway